tests/sentry/integrations/github_enterprise/test_ticket_action.py

```diff
@@ -0,0 +1,193 @@
+from unittest.mock import patch
+
+import pytest
+import responses
+from django.urls import reverse
+from rest_framework.test import APITestCase as BaseAPITestCase
+
+from sentry.eventstore.models import Event
+from sentry.integrations.github.integration import GitHubIntegration
+from sentry.integrations.github_enterprise import GitHubEnterpriseCreateTicketAction, client
+from sentry.models.integrations.external_issue import ExternalIssue
+from sentry.models.rule import Rule
+from sentry.testutils.cases import RuleTestCase
+from sentry.testutils.skips import requires_snuba
+from sentry.types.rules import RuleFuture
+
+pytestmark = [requires_snuba]
+
+
+class GitHubEnterpriseEnterpriseTicketRulesTestCase(RuleTestCase, BaseAPITestCase):
+    rule_cls = GitHubEnterpriseCreateTicketAction
+    repo = "foo/bar"
+    assignee = "sentry_user"
+    labels = ["bug", "invalid"]
+    issue_num = 1
+
+    def setUp(self):
+        super().setUp()
+        self.integration = self.create_integration(
+            organization=self.organization,
+            provider="github_enterprise",
+            name="Github Enterprise",
+            external_id="1",
+            metadata={
+                "domain_name": "github.example.com",
+                "installation": {
+                    "private_key": "fake key",
+                    "id": 1,
+                    "verify_ssl": True,
+                },
+                "installation_id": 1234,
+            },
+        )
+
+        self.installation: GitHubIntegration = self.integration.get_installation(self.organization.id)  # type: ignore[assignment]
+
+        self.login_as(user=self.user)
+        responses.add(
+            method=responses.POST,
+            url="https://github.example.com/api/v3/app/installations/1234/access_tokens",
+            body='{"token": "12345token", "expires_at": "2099-01-01T00:00:00Z"}',
+            status=200,
+            content_type="application/json",
+        )
+
+    @pytest.fixture(autouse=True)
+    def stub_get_jwt(self):
+        with patch.object(client, "get_jwt", return_value="jwt_token_1"):
+            yield
+
+    def trigger(self, event, rule_object):
+        action = rule_object.data.get("actions", ())[0]
+        action_inst = self.get_rule(data=action, rule=rule_object)
+        results = list(action_inst.after(event=event, state=self.get_state()))
+        assert len(results) == 1
+
+        rule_future = RuleFuture(rule=rule_object, kwargs=results[0].kwargs)
+        return results[0].callback(event, futures=[rule_future])
+
+    def get_key(self, event: Event):
+        return ExternalIssue.objects.get_linked_issues(event, self.integration).values_list(
+            "key", flat=True
+        )[0]
+
+    @responses.activate()
+    def test_ticket_rules(self):
+        title = "sample title"
+        sample_description = "sample bug report"
+        html_url = f"https://github.example.com/api/v3/foo/bar/issues/{self.issue_num}"
+
```

Comment:
I'm surprised we don't need a response like this: https://github.com/getsentry/sentry/blob/master/tests/sentry/integrations/github_enterprise/test_issues.py#L117-L121 but I haven't worked w/ GHE much

Reply:
🤞

Reply:
Wait turns out I am doing it lol: https://github.com/getsentry/sentry/pull/59315/files#diff-48d5259d445171b0a9fc4a47007fb56b42dcee4decf40b649f3b2b97f6ba2bddR56-R59